### ISSUE TITLE
Detecting GraphQL schema files in a generated directory?

### ISSUE THREAD

**Issue by eburke56** [jimkyndemeyer/js-graphql-intellij-plugin#493]
We are trying to set up our Android Studio project so that the plugin can detect our schema.  Because the schema lives in another artifact, we set up a Gradle task to copy the schema to our codegen directory.  However, the plugin does not see this file, presumably because it is not a "project file".

Is there a way we can inform the plugin that it can find the file in a directory under `build/generatedGraphQLData/schema`?  I've tried configuring a `.graphqlconfig` with no luck.

I created a symlink from the toplevel directory `schema -> build/generatedGraphQLData/schema`, and it works.

**Comment by vepanimas** [comment]
@eburke56 hi! Do you have any .graphqlconfig files? Could you share them if so? I've tried to reproduce it, but in my case the schema from the generated root is successfully resolved in queries from the other modules. Maybe your project structure is more complicated than my example project. Also what plugin and IDE version do you use?

**Comment by vepanimas** [comment]
That's my example empty project, you can see that a `query.graphql` found a schema file in the generated root even without a provided `.graphqlconfig`.

<img width="623" alt="gen-roots" src="https://user-images.githubusercontent.com/10243762/133683633-e3ea9fa6-5f7c-47ce-8647-d67d499ea2ea.png">

**Comment by eburke56** [comment]
Thanks for the reply!  I think the difference is that my directory is not seen by AS as a "generated sources root".  If I manually mark the directory as a sources root, it works.

**Comment by vepanimas** [comment]
@eburke56 was it marked as excluded before in your case? If so, you could simply do "Unmark as excluded" or "Mark as source directory". Or "Mark as generated root" should help too.

**Comment by eburke56** [comment]
Yes, it was marked as excluded after being generated.  I'm trying to find a way to have the Gradle task mark it as such automatically, e.g., by adding it to a sources root.

**Comment by vepanimas** [comment]
@eburke56 Have you managed to fix it on your side? I'm not a Gradle pro, but I can still ask the Gradle integration plugin maintainers about it if it's still a problem for you.

**Comment by mertdurukan** [comment]
As there is no "Mark directory as" menu in Android Studio, I found out that such settings like including and excluding directories are stored in a ".iml" file. Named like "project-name.iml".

A line like bellow is used to exclude a dir:

`<excludeFolder url="file://$MODULE_DIR$/src/main/x" />`
So, Re-including is done as easily as removing the line.
